code/drasil-code/Language/Drasil/Code/Imperative/Helpers.hs

```diff
@@ -15,7 +15,7 @@ liftS :: State a b -> State a [b]
 liftS = fmap (: [])
 
 -- | For an expression using a less than operator, gets the upper bound.
-getUpperBound :: Expr -> Expr
+getUpperBound :: CodeExpr -> CodeExpr
```

Comment:
Can you open an issue to get rid of this function? We shouldn't doing 'introspection' like this.

Reply:
Ah, yes, we had this issue before actually. #2414 is the related ticket, we only have 1 usage of it -- and it's very small and in `drasil-code`, I think we can add it to the scope of this PR too.

Reply:
There's a few too many to add on, I actually confused `getUpperBound` above with another, `getInnerSpace` which I used in an example. Let's leave it to another PR -- I'll try to get it done today however.